sdk/dotnet/Pulumi/Serialization/Attributes.cs

```diff
@@ -94,4 +94,17 @@ public sealed class OutputConstructorAttribute : Attribute
     public sealed class EnumTypeAttribute : Attribute
     {
     }
+    
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class ResourceIdentifierAttribute : Attribute
```

Comment:
Nit: Any reason why we wouldn't name it simply `ResourceAttribute`? Or `ResourceTypeAttribute`?

Reply:
I believe `[Resource()]` will confuse the compiler vs. `Pulumi.Resource` (and maybe confusing for humans?) `ResourceTypeAttribute` is probably a better name indeed.